trio/_core/tests/test_watchdog.py

```diff
@@ -0,0 +1,30 @@
+import threading
+import time
+from io import StringIO
+
+import contextlib
+
+from ..tests.tutil import slow
+from .. import _run
+
+
+@slow
+async def test_watchdog():
+    watchdog = _run.GLOBAL_RUN_CONTEXT.watchdog
+    target = StringIO()
+    with contextlib.redirect_stderr(target):
+        time.sleep(7)
+
```

Comment:
Maybe it would also be good to have a test where we `await trio.sleep(7)` and make sure the watchdog doesn't fire? (I.e., a test to catch the bug that was in the first draft of this code.)

Also, these sleeps are kind of long, even for a `@slow` test... maybe it would be nicer to make the argument `watchdog_timeout=5`, so then in tests we can adjust it? (And we get rid of the unconfigurable magic constant.)